# Palmer Penguins

![](https://upload.wikimedia.org/wikipedia/commons/thumb/6/69/Manchot_01.jpg/440px-Manchot_01.jpg)

Este conjunto de datos contiene sobre 3 especies de pinguinos que habitan en las islas Palmer. 

## Libreria

In [3]:
# Tratamiento de datos
import numpy as np
import pandas as pd

#graficas
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objects as go

#estadistiuca
import scipy

In [6]:
pinguino=pd.read_csv(r'C:\Users\denis\Desktop\Curso\Juan-Temario\DS102024_Nuevo\2-DataAnalysis\10-EDA\penguins.csv')

## Primera Exploracion

In [7]:
pinguino.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     342 non-null    float64
 4   bill_depth_mm      342 non-null    float64
 5   flipper_length_mm  342 non-null    float64
 6   body_mass_g        342 non-null    float64
 7   sex                333 non-null    object 
 8   year               344 non-null    int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 24.3+ KB


In [8]:
pinguino.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [10]:
pinguino.duplicated().sum()

np.int64(0)

In [15]:
pinguino.describe(include='all')

rowid species  island  bill_length_mm  bill_depth_mm  \
count   344.000000     344     344      342.000000     342.000000   
unique         NaN       3       3             NaN            NaN   
top            NaN  Adelie  Biscoe             NaN            NaN   
freq           NaN     152     168             NaN            NaN   
mean    172.500000     NaN     NaN       43.921930      17.151170   
std      99.448479     NaN     NaN        5.459584       1.974793   
min       1.000000     NaN     NaN       32.100000      13.100000   
25%      86.750000     NaN     NaN       39.225000      15.600000   
50%     172.500000     NaN     NaN       44.450000      17.300000   
75%     258.250000     NaN     NaN       48.500000      18.700000   
max     344.000000     NaN     NaN       59.600000      21.500000   

        flipper_length_mm  body_mass_g   sex         year  
count          342.000000   342.000000   333   344.000000  
unique                NaN          NaN     2          NaN  
top                   NaN          NaN  male          NaN  
freq                  NaN          NaN   168          NaN  
mean           200.915205  4201.754386   NaN  2008.029070  
std             14.061714   801.954536   NaN     0.818356  
min            172.000000  2700.000000   NaN  2007.000000  
25%            190.000000  3550.000000   NaN  2007.000000  
50%            197.000000  4050.000000   NaN  2008.000000  
75%            213.000000  4750.000000   NaN  2009.000000  
max            231.000000  6300.000000   NaN  2009.000000

In [18]:
print(pinguino['species'].unique())
print(pinguino['island'].unique())
print(pinguino['sex'].unique())

['Adelie' 'Gentoo' 'Chinstrap']
['Torgersen' 'Biscoe' 'Dream']
['male' 'female' nan]


### como lo haria Dani

### conclusion

1. se puede observar que los dtos estan compuestos por 344 filas y 9 columnas de las cuales ´rowid´ carece de importancia ya que es un indice.

|Variable|Descripción|
|---|---|
|Especie (species)| Adelie, Chinstrap, Gentoo.|
|Isla (island)| Biscoe, Dream, Torgersen.|
|Longitud del pico (bill_length_mm)| medida en milímetros.|
|Profundidad del pico (bill_depth_mm)| medida en milímetros.|
|Longitud de la aleta (flipper_length_mm)| medida en milímetros.|
|Masa corporal (body_mass_g)| medida en gramos.|
|Género (sex)| Macho o Hembra.|
|Año (year)| Año en que se recolectaron los datos.|

## Analisis Exploratorio de Datos (EDA)

### Analisis Univariante